lnwire/neighbor_hello.go

```diff
@@ -5,29 +5,24 @@
 package lnwire
 
 import (
-	"encoding/gob"
 	"fmt"
 	"io"
 
 	"github.com/BitfuryLightning/tools/rt"
 )
 
 type NeighborHelloMessage struct {
-	RoutingMessageBase
 	RT *rt.RoutingTable
 }
 
 func (msg *NeighborHelloMessage) Decode(r io.Reader, pver uint32) error {
-	decoder := gob.NewDecoder(r)
-	rt1 := rt.NewRoutingTable()
-	err := decoder.Decode(rt1.G)
+	rt1, err := rt.UnmarshallRoutingTable(r)
```

Comment:
The encode/decode methods for all the newly added routing structs should be implemented using raw binary serialization in the same manner the other messages within `lnwire` are. Gob shouldn't be used underneath as the serialization format isn't portable across languages other than Go. 

This currently isn't a blocker for this PR, but we should resolve this within the forthcoming PR which integrates the routing package itself into the project.